### PR TITLE
Enhancement according to Issue #1

### LINKED ISSUE #1: Modifications to Controller
Suggesting below changes in the Controller:
- Routes for the controller must start with the entity which would be getting updated
  - For example, a GET API call to get one customer should be something like `customer/{id}`. `get` need not be in the route itself, since method name is sufficient indication for the endpoint behaviour (Links for further reading: [Link-1](https://nordicapis.com/10-best-practices-for-naming-api-endpoints/), [Link-2](https://restfulapi.net/resource-naming/)
- Better response messages: API Response can give more details about the entity that was updated (probably id of the created/updated/deleted entry).

### PR DESCRIPTION
**Pull request by sauravp1**
Issues #1 

Changed Endpoints:

I have changed endpoints according to guidelines
i have removed verbs from uri and now only http verbs can help in crud
Now you can simply use :
https://www.example.com/customer with POST and GET to (add customer and view all customer)
https://www.example.com/customer/{id} will perform update, delete, and get single order according to HTTP VERBS used.

Changed Response:

Now crud will return id of modified objects.

### CODE REVIEW

File: src/Controller/CustomerController.php [hunk 1]
```diff
@@ -44,14 +44,14 @@ public function addCustomer(Request $request): JsonResponse
 
          }
 
-         $this->customerRepository->saveCustomer($firstName, $lastName, $email, $phoneNumber);
-         return new JsonResponse(['status' => "Customer created!"], Response::HTTP_CREATED);
+         $id = $this->customerRepository->saveCustomer($firstName, $lastName, $email, $phoneNumber);
+         return new JsonResponse(['status' => "Customer created with id number {$id}"], Response::HTTP_CREATED);
```
Comment (srinidhibhat-practo) [inline src/Controller/CustomerController.php]: `status` might not be a very good choice for naming the response message key. 
Maybe `message` would suffice?

Same is applicable for all usages in this file.

File: src/Controller/CustomerController.php [hunk 2]
```diff
@@ -91,7 +91,7 @@ public function getAllCustomers(): JsonResponse
 
 
     /**
-     * @Route("/update/{id}", name="update_customer", methods={"PUT"})
+     * @Route("/{id}", name="update_customer", methods={"PUT"})
```
Comment (srinidhibhat-practo) [inline src/Controller/CustomerController.php:94]: I think `PATCH` makes more sense here. `PUT` is often used to update entire resource ([reference](https://www.baeldung.com/http-put-patch-difference-spring))
I'm assuming `PUT` was considered just done for the demo purpose.

Same is applicable for other usages of `PUT` in this file

**Review by srinidhibhat-practo — COMMENTED** [review]
@sauravp1 Added couple of comments.

I believe there is a missing call to `persist` the entity in `customerRepository->updateCustomer()`
Can you verify?

**Comment by sauravp1** [comment]
> @sauravp1 Added couple of comments.
> 
> I believe there is a missing call to `persist` the entity in `customerRepository->updateCustomer()` Can you verify?

Persist is basically used when creating new entry and we can ignore persist() during update because doctrine knows which object to check for modification and what operation to perform

https://symfonycasts.com/screencast/symfony-doctrine/update#no-persist-on-update

File: src/Controller/CustomerController.php (hunk 2 above)
Comment (sauravp1) [reply src/Controller/CustomerController.php:94]: I have used `PATCH` for demo purpose only.

File: src/Controller/CustomerController.php (hunk 1 above)
Comment (sauravp1) [reply src/Controller/CustomerController.php]: naming response changed from `status` to `message`

**Review by srinidhibhat-practo — APPROVED** [review]

**Pull request merged**